Client/Frontend/Sync/SyncViewController.swift

```diff
@@ -43,4 +49,10 @@ class SyncViewController: UIViewController {
         
         code()
     }
+    
+    @objc func didLeaveSyncGroup() {
+        DispatchQueue.main.async { [weak self] in
+            self?.dismiss(animated: true, completion: nil)
```

Comment:
Wouldn't this pop the entire settings page?

Generally this should go inside the VC that is actually presenting, but this is probably fine here.

Reply:
Yes it drops the whole settings VC. I didn't want to implement it in all of the sync screens for sake of simplicity.
Device removal can happen in few of those Sync view controllers

Reply:
Understood, wdyt about having SettingsViewControl observe this though, since that is the one presenting the sync screens, and do something like:

```
didLeaveSyncGroup() {
    if children.contains(where: { $0.isKind(of: SyncViewController.classForCoder()) }) {
        self.navigationController.popToViewController(self, animated: true)
    }
}
```

This is a bit cleaner separation of concerns, since the child vc's do not need to worry about how they are being presented.

Reply:
`navigationController` is nil in settings vc when another vc is presented. Please see my updated solution now, it should fix the problem.